zk/stages/stage_sequence_execute.go

```diff
@@ -98,36 +102,40 @@ func SpawnSequencingStage(
 	var decodedBlocks []zktx.DecodedBatchL2Data // only used in l1 recovery
 	var blockTransactions []types.Transaction
 	var effectiveGases []uint8
+	l1Recovery := cfg.zk.L1SyncStartBlock > 0
```

Comment:
do you think we should also have an explicit flag about it being an L1 recovery mode?

Reply:
That flag triggers the recovery.  In theory you could leave it like that forever and it would just keep syncing from the L1 so it's more of an alternative syncing method rather than an explicit recovery.  Maybe rename it as such instead?  like `l1SyncMode` rather than `l1Recovery`, wdyt?

Reply:
I mostly was thinking to have one extra flag to be explicit on triggering the recovery, because the block being > 0 is a bit strange, block 0 is a perfectly valid block. very unlikely, but still.